packages/cli/src/commands/dev.ts

```diff
@@ -31,10 +32,21 @@ const defaultIgnored = [
 
 const devAbortController = new AbortController()
 
-async function storeDev(rootDir: string, tmpDir: string) {
+async function storeDev(rootDir: string, tmpDir: string, coreDir: string) {
   const envVars = dotenv.parse(readFileSync(path.join(rootDir, 'vtex.env')))
 
-  const devProcess = spawn('yarn dev', {
+  runCommandSync({
+    cmd: 'yarn predev',
+    errorMessage:
+      'GraphQL was not optimized and TS files were not updated. Changes in the GraphQL layer did not take effect',
```

Comment:
question: here other commands are run, right? `yarn generate:schema && yarn generate:codegen && yarn format:generated` not only the `yarn generate:codegen`. What would happen if only the `format` returned error? It would fail but the previous commands would have made changes on the files?

Reply:
Yes, but the changes are made to ephemeral files so, theoretically, no harm is done. Because of `throws: 'error'` here, the behavior is the same. Whatever step fails there will be no rollback.